src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -22,8 +29,21 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         }
         else
         {
-            std::hash<IActionArgs> argsHash;
-            hashedArgs = argsHash(nullptr);
+            // Check if this action is supposed to have args.
+            Model::IActionArgs hiddenActionArgs{ nullptr };
+#define ON_ALL_ACTIONS_WITH_ARGS(action) CHECK_IF_ACTION_WITH_ARG(actionAndArgs.Action(), action, hiddenActionArgs)
+            ALL_SHORTCUT_ACTIONS_WITH_ARGS
+#undef ON_ALL_ACTIONS_WITH_ARGS

```

Comment:
This can be implemented using a neat, efficient switch/case:
```cpp
switch (actionAndArgs.Action())
{
#define ON_ALL_ACTIONS_WITH_ARGS(action) case action: hiddenActionArgs = winrt::make<action##Args>(); break;
    ALL_SHORTCUT_ACTIONS_WITH_ARGS
#undef ON_ALL_ACTIONS_WITH_ARGS
}
```